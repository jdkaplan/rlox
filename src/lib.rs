macro_rules! debug_log_gc {
    ($($arg:tt)*) => {{
        #[cfg(feature = "log_gc")]
        println!($($arg)*);
    }};
}

mod chunk;
mod compiler;
mod gc;
mod heap;
mod object;
mod scanner;
mod table;
mod value;
mod vm;

pub use vm::{InterpretError, InterpretResult, Vm};

pub const U8_COUNT: usize = (u8::MAX as usize) + 1;
pub const FRAMES_MAX: usize = 64;
pub const STACK_MAX: usize = FRAMES_MAX * U8_COUNT;
