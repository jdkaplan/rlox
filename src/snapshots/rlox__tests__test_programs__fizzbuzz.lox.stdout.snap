---
source: src/lib.rs
description: "fun div(n, m) {\n    // TODO: modulo operator\n    while (n > 0 and n >= m) {\n        n = n - m;\n    }\n    return n == 0;\n}\n\nfor (var i = 1; i <= 30; i = i + 1) {\n    if (div(i, 15)) print \"fizzbuzz\";\n    else if (div(i,  3)) print \"fizz\";\n    else if (div(i,  5)) print \"buzz\";\n    else print i;\n}\n"
input_file: test_programs/fizzbuzz.lox
---
1
2
fizz
4
buzz
fizz
7
8
fizz
buzz
11
fizz
13
14
fizzbuzz
16
17
fizz
19
buzz
fizz
22
23
fizz
buzz
26
fizz
28
29
fizzbuzz

